termd-build-driver/src/main/java/org/jboss/pnc/termdbuilddriver/transfer/DefaultFileTranser.java

```diff
@@ -152,4 +181,64 @@ public void setConnectTimeout(int connectTimeout) {
     public void setReadTimeout(int readTimeout) {
         this.readTimeout = readTimeout;
     }
+
+    public void setHttpRetryMaxAttempts(int httpRetryMaxAttempts) {
+        this.httpRetryMaxAttempts = httpRetryMaxAttempts;
+    }
+
+    public int getHttpRetryMaxAttempts() {
+        return httpRetryMaxAttempts;
+    }
+
+    public void setHttpRetryWaitBeforeRetry(long httpRetryWaitBeforeRetry) {
+        this.httpRetryWaitBeforeRetry = httpRetryWaitBeforeRetry;
+    }
+
+    public long getHttpRetryWaitBeforeRetry() {
+        return httpRetryWaitBeforeRetry;
+    }
+
+    /**
+     * Helper method to sleep the thread for x milliseconds. Written so that we don't have to deal with exceptions
+     *
+     * @param milliseconds amount of sleep
+     */
+    private void sleep(long milliseconds) {
+        try {
+            Thread.sleep(httpRetryWaitBeforeRetry);
+        } catch (InterruptedException f) {
+            logger.warn("Could not sleep peacefully!");
+        }
+    }
+
+    /**
+     * Retry to run the myRunnable if a transfer exception is thrown, up until httpRetryMaxAttempts. Sleep for
+     * httpRetryWaitBeforeRetry milliseconds between retries.
+     *
+     * If we have reached the max number of attempts, rethrow the transfer exception
+     *
+     * @param myRunnable task to run
+     * @param taskName task name to use for logging
+     * @throws TransferException thrown if the task fails after httpRetryMaxAttempts
+     */
+    <T> T retry(MyRunnable<T> myRunnable, String taskName) throws TransferException {
+
+        for (int i = 1; i <= httpRetryMaxAttempts; i++) {
+
+            try {
+                return myRunnable.run();
+            } catch (TransferException e) {
+                // if we have reached the max number of attempts, give up and throw the transfer exception
+                if (i == httpRetryMaxAttempts) {
+                    throw e;
+                } else {
+                    logger.warn("{} failed after {} attempts. Sleeping...", taskName, 1, e);
```

Comment:
It probably should be "i" instead of "1"

Reply:
Done! Thanks for spotting this! Haa silly bug :)